### PR TITLE
Add limited GA to the buyers guide

### PR DESCRIPTION
**Pull request by Pomax**
Addresses some of the events in https://github.com/mozilla/foundation.mozilla.org/issues/1925, but not done in a real sense.

Added:
- support link click event
- "copy link" click event

note that bg-main.js does not import and init `product-analytics.js` yet

### CODE REVIEW

**Comment by Pomax** [comment]
@alanmoo @mmmavis hopefully this is enough to just finish up on

File: network-api/networkapi/buyersguide/templates/social.html [hunk 1]
```diff
@@ -9,10 +9,10 @@
...
-  <a class="social-button social-button-twitter hidden-sm-down" href="{{twitter_share_link}}{{share_text|urlencode}}" target="_blank"><span class="sr-only">Twitter</span></a>
-  <a class="social-button social-button-email hidden-sm-down" href="{{email_share_link}}{{share_text|urlencode}}"><span class="sr-only">Email</span></a>
-  <a class="social-button social-button-link copy-link hidden-sm-down" href=""><span class="sr-only">Copy link</span></a>
+  <a class="social-button social-button-fb hidden-sm-down" id="nav-social-button-fb" href="{{facebook_share_link}}"><span class="sr-only">Facebook</span></a>
+  <a class="social-button social-button-twitter hidden-sm-down" id="nav-social-button-twitter" href="{{twitter_share_link}}{{share_text|urlencode}}" target="_blank"><span class="sr-only">Twitter</span></a>
+  <a class="social-button social-button-email hidden-sm-down" id="nav-social-button-email" href="{{email_share_link}}{{share_text|urlencode}}"><span class="sr-only">Email</span></a>
+  <a class="social-button social-button-link copy-link hidden-sm-down" id="nave-social-button-link" href=""><span class="sr-only">Copy link</span></a>
```
Comment (mmmavis) [inline network-api/networkapi/buyersguide/templates/social.html]: `nave` typo?

Comment (mmmavis) [reply network-api/networkapi/buyersguide/templates/social.html]: Are we tracking clicks on mobile as well? If so let's add `id` to elements on mobile version.

File: source/js/buyers-guide/product-analytics.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    'product-copy-link-button': {
+      category: `product`,
+      action: `copy link tap`,
+      label: `copy link ${productName}`
+    },
+    'creep-vote': {
+      category: `product`,
+      action: `opinion submitted`,
+      label: `opinion on ${productName}`
+    },
```
Comment (mmmavis) [inline source/js/buyers-guide/product-analytics.js]: - panning Creep-O-Meter seems to trigger this event as well. I'm not sure if this is the desired event we wanna attach to this action
image

- answering "How likely are you to buy it?" double trigger this event
image

File: network-api/networkapi/buyersguide/templates/primary_nav.html [hunk 3]
```diff
@@ -11,7 +11,7 @@
                   <div class="col">
                     <h3 class="h3-heading">Help us keep the Internet healthy</h3>
                     <p>Mozilla is a nonprofit organization fighting for a healthy interent.</p>
-                    <a class="btn btn-donate m-0 w-100" href="https://donate.mozilla.org">Donate</a>
+                    <a class="btn btn-donate m-0 w-100" id="donate-button" href="https://donate.mozilla.org">Donate</a>
```
Comment (mmmavis) [inline network-api/networkapi/buyersguide/templates/primary_nav.html]: `id` needs to be added to desktop version element as well

File: source/js/buyers-guide/product-analytics.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import ReactGA from 'react-ga';
+
+function getElementGAInformation(pageTitle, productName) {
+  return {
+    'product-live-chat`': {
```
Comment (mmmavis) [inline source/js/buyers-guide/product-analytics.js]: extra ` here

File: source/js/buyers-guide/product-analytics.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      category: `product`,
+      action: `copy link tap`,
+      label: `copy link ${productName}`
+    },
+    'creep-vote': {
+      category: `product`,
+      action: `opinion submitted`,
+      label: `opinion on ${productName}`
+    },
+    'product-social-button-fb': {
```
Comment (mmmavis) [inline source/js/buyers-guide/product-analytics.js]: maybe these are todos for later - we need to assign elements with the following ids still `product-social-button-fb`, `product-social-button-twitter`, and `product-social-button-email`

File: network-api/networkapi/buyersguide/templates/social.html (hunk 1 above)
Comment (Pomax) [reply network-api/networkapi/buyersguide/templates/social.html]: :+1: let me add those

File: source/js/buyers-guide/product-analytics.js (hunk 2 above)
Comment (Pomax) [reply source/js/buyers-guide/product-analytics.js]: looks like this was tacked onto "the form" which means everytime you click the form it fires! adding an `id` to the vote button and fixing it to use that instead. Good catch!

File: network-api/networkapi/buyersguide/templates/social.html [hunk 6]
```diff
@@ -2,18 +2,18 @@
...
+    <a class="social-button social-button-link copy-link body-large" id="nav-social-button-link" href="">Copy link</a>
   </div>
 {% else %}
 <div class="social d-flex align-items-center">
-  <a class="social-button social-button-fb hidden-sm-down" href="{{facebook_share_link}}"><span class="sr-only">Facebook</span></a>
-  <a class="social-button social-button-twitter hidden-sm-down" href="{{twitter_share_link}}{{share_text|urlencode}}" target="_blank"><span class="sr-only">Twitter</span></a>
-  <a class="social-button social-button-email hidden-sm-down" href="{{email_share_link}}{{share_text|urlencode}}"><span class="sr-only">Email</span></a>
-  <a class="social-button social-button-link copy-link hidden-sm-down" href=""><span class="sr-only">Copy link</span></a>
-  <a class="d-none d-sm-block btn btn-donate ml-3 mr-0 hidden-sm-down" href="https://donate.mozilla.org">Donate</a>
+  <a class="social-button social-button-fb hidden-sm-down" id="nav-social-button-fb" href="{{facebook_share_link}}"><span class="sr-only">Facebook</span></a>
```
[the remark this answers is not in the document]
Comment (mmmavis) [reply network-api/networkapi/buyersguide/templates/social.html]: Good catch. I forgot we don't conditionally render the elements DOM based on the screen size, we just toggle their display state from hidden to visible. Same `id` will create problems!

[the remark this answers is not in the document]
Comment (Pomax) [reply network-api/networkapi/buyersguide/templates/social.html]: added `-mb` suffixes for mobile, with GA rules for them in the analytics js file

**Pull request merged**